woocommerce/woocommerce-gutenberg-products-block
# Add DELETE `private/ai/pattern` endpoint

## What

Fixes #11873

## Why



We have to add the DELETE `/pattern` endpoint to allow to remove the patterns generated by the AI via REST API.

## Testing Instructions

### Setup

Please test the feature on a WooExpress install instead of Jurassic Ninja, as the latter can be very limited in terms of server resources and unstable, as other installs can interfere with your own.

- If you already have a pre-configured install, remove the pre-existing WooCommerce Blocks Plugin and upload the one from this zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/13440451/woocommerce-gutenberg-products-block.zip)
- Alternatively, create a new WooExpress install from scratch 
- Remove the pre-existing WooCommerce plugin via SSH and
- Install and activate WooCommerce from the following zip file: 
[woocommerce.zip](https://github.com/woocommerce/woocommerce-blocks/files/13415324/woocommerce.zip)
- Install and activate WooCommerce Blocks from the following zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/13440451/woocommerce-gutenberg-products-block.zip)
- Install and activate WooCommerce Beta Tester (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

<img width="1233" alt="Screenshot 2023-10-22 at 10 32 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/816867c6-34e5-40a1-a87d-f4a7cda46429">

### Test the CYS experience

0. Make sure you don't have any products in your store.
1. Head over to Home > Customize your Store:
 
<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/b5d34131-3edc-47ed-a8e4-8e2775ca1bbd">

2. Click on design with AI.
3. Type a description for your business.
4. Ensure that the generation of the pattern is terminated.
5. Connect to the database and ensure that there is a post with the post_type `patterns_ai_data`.
6. Create a request with the method DELETE on the endpoint `/wp-json/wc/private/ai/patterns`.
7. Connect to the database and ensure that there isn't any post with the post_type `patterns_ai_data`.



## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
